### ISSUE TITLE
Unmarshalling string content blocks fails

### ISSUE THREAD

**Issue by tonyhb** [liushuangls/go-anthropic#58]
According to the docs:

> Each input message content may be either a single string or an array of content blocks, where each block has a specific type. Using a string for content is shorthand for an array of one content block of type "text". The following input messages are equivalent:

```{"role": "user", "content": "Hello, Claude"}```
```{"role": "user", "content": [{"type": "text", "text": "Hello, Claude"}]}```

Unmarshalling a valid string message currently fails:

```
json: cannot unmarshal string into Go struct field Message.messages.content of type []anthropic.MessageContent
```

I'll put up a patch to handle unmarshalling in the string case.

**Comment by liushuangls** [comment]
Yes, it's written in the documentation that content can be shortened to a string, and I only support array structures for consistency

**Comment by liushuangls** [comment]
Are you saying that claude's response might have a string content? I've never encountered that
